### DISCUSSION TITLE
Pymatgen Structure as_dict(), from_dict()

### DISCUSSION THREAD

**Discussion by anyangml** [materialsproject/pymatgen#3925] [Q&A]
I have a dataset of relaxation trajectories. I would like to compare the coordinates before and after relaxation by calculating the RMSE. I thought it's very trivial, I can simply iterate over all sites and all x,y,z to compare. The problem is that the relaxed structure may have some `rotation`,`permutation`,`translation` operations applied to them, such that I can not directly compare the coordinates.
for example, in the image below, each row represents an atom, and columns are the x, y, z. It seems y and z are flipped and atoms are permuted.
![20240716-201815](https://github.com/user-attachments/assets/6e9a9fd5-41be-4168-a908-842ba162e741)

I wonder if it's because I used `as_dict()`,`from_dict` to save the relaxed structures.
```python
AseAtomsAdaptor.get_structure(atoms).as_dict()
```
Is there a way to standardize the structure so that I can directly compare x,y,z?
I tried StructureMatcher.get_s2_like_s1(), but it seems the returned coordinates still differ significantly.

Any suggestion would be greatly appreciated!

**Comment by DanielYang59** [comment]
Hi @anyangml  That sounds like an interesting question.

> The problem is that the relaxed structure may have some rotation,permutation,translation operations applied to them, such that I can not directly compare the coordinates.

I’m unsure which package was used to generate these trajectories, but isn't it possible that these coordinates are the final state with all transformations already applied?

> each row represents an atom, and columns are the x, y, z. It seems y and z are flipped and atoms are permuted.

Wondering how you got to the conclusion that certain axes/atoms have been swapped? Perhaps you could provide the original file if possible?
